wvm-apps/wvm-exexed/crates/precompiles/src/inner/arweave_read_precompile.rs

```diff
@@ -0,0 +1,146 @@
+use serde::{Deserialize, Serialize};
+use reth::primitives::Bytes;
+use reth::primitives::revm_primitives::{Precompile, PrecompileError, PrecompileErrors, PrecompileOutput, PrecompileResult};
+use reth::revm::precompile::{PrecompileWithAddress, u64_to_address};
+
+pub const PC_ADDRESS: u64 = 0x18;
+pub const ARWEAVE_PC_READ_BASE: u64 = 3_450;
+
+pub const TX_MAX_SIZE: usize = 18_874_368; // 18MB
+
+pub const ARWEAVE_READ_PC: PrecompileWithAddress =
+    PrecompileWithAddress(u64_to_address(PC_ADDRESS), Precompile::Standard(arweave_read));
+
+pub const ARWEAVE_TX_ENDPOINT: &str = "https://arweave.net/";
+
+pub const ARWEAVE_GRAPHQL_ENDPOINT: &str = "https://arweave.net/graphql";
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Response {
+    data: Data,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Data {
+    transactions: Transactions,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Transactions {
+    edges: Vec<Edge>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct Edge {
+    node: Node,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct Node {
+    id: String,
+    data: NodeData,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct NodeData {
+    size: String,
+}
+
+fn arweave_read(input: &Bytes, gas_limit: u64) -> PrecompileResult {
+    let data_size = input.len();
+    let gas_used: u64 = (10_000 + data_size * 3) as u64;
```

Comment:
update `ARWEAVE_PC_READ_BASE` to 10_000 and use the constant variable instead of hardcoded base gas value